### PR TITLE
Fix/replace shortuuid dependency with code

### LINKED ISSUE #84: Reduce dependencies
Our current dependencies were chosen out of habit and convenience without considering that this is a library so we should have a minimal set of dependencies:

- [x] Raise an exception with a clear message to install fastapi whenever `spylib/oauth/router.py` is loaded and fastapi cannot be imported.
- [x] Remove loguru and use the standard library logger instead.
- [x] Copy paste code from the shortuuid library that we need to make our `get_unique_id` independent of shortuuid
- [x] Move nest-asyncio to dev dependencies (must have been put as main dependency by accident)

### PR DESCRIPTION
**Pull request by lishanl**
Fix #84 
Copy over shortuuid code used in `get_unique_id()`

@hillairet Do you have on mind how to credit to the shortuuid lib? Thanks

### CODE REVIEW

**Comment by hillairet** [comment]
@lishanl Good question!
A comment in the code would be a good start.

@qw-in Do you know what is the best practice when grabbing code from a library like this?

**Comment by hillairet** [comment]
@jsemearns I just remembered that we moved this to toolip.
We at least need toolip to also copy over the code. I don't think we want toolip as a dependency in SPyLib though

**Review by hillairet — APPROVED** [review]
Not much code to get rid of a dependency! Sweet!

**Comment by qw-in** [comment]
@hillairet not sure - I usually just add a permalink to the code I copied. Could also put in a vendor's folder but that is likely overkill

File: spylib/utils/constants.py [hunk 1]
```diff
@@ -0,0 +1 @@
+ALPHABET = '23456789ABCDEFGHJKLMNPQRSTUVWXYZabcdefghijkmnopqrstuvwxyz'
```
Comment (qw-in) [inline spylib/utils/constants.py]: Why does the alphabet start at 2?

I'd suggest keeping all the code copied from an external library all in one file or folder

**Comment by hillairet** [comment]
> @hillairet not sure - I usually just add a permalink to the code I copied. Could also put in a vendor's folder but that is likely overkill

Cool so a simple comment "Code copied over from shortuuid package: <PERMALINK>" would be fine.
Then having this at the top of the file (called shortuuid???) would be logical.

File: spylib/utils/constants.py (hunk 1 above)
Comment (lishanl) [reply spylib/utils/constants.py]: I wondered that too, but it's just copied from the code https://github.com/skorokithakis/shortuuid/blob/master/shortuuid/main.py#L42

**Review by qw-in — APPROVED** [review]

**Review by hillairet — APPROVED** [review]
A good thing done!

I really wonder about this missing 1 in the alphabet.
Damnit @qw-in! Now I need to know!  :wink: 
@lishanl If you don't mind looking a little bit that'd be lovely but don't spend too much time on this.

**Comment by lishanl** [comment]
> A good thing done!
> 
> I really wonder about this missing 1 in the alphabet. Damnit @qw-in! Now I need to know! 😉 @lishanl If you don't mind looking a little bit that'd be lovely but don't spend too much time on this.

🕵🏼‍♀️

**Comment by lishanl** [comment]
> A good thing done!
> 
> I really wonder about this missing 1 in the alphabet. Damnit @qw-in! Now I need to know! 😉 @lishanl If you don't mind looking a little bit that'd be lovely but don't spend too much time on this.

I hope the answer by the Author doesn't disappoint you. 😉 
https://github.com/skorokithakis/shortuuid/pull/33#issuecomment-313862131
@hillairet @qw-in

**Pull request merged**